hermes_memory_provider/__init__.py

```
@@ -1531,16 +1577,67 @@ def _should_filter(self, content: str) -> bool:
         """Read a single key from memory.mnemosyne in config.yaml."""
         try:
-            import yaml, os
+            import os
             config_path = os.path.join(self._hermes_home, "config.yaml") if self._hermes_home else ""
             if not config_path or not os.path.exists(config_path):
                 return None
+            try:
+                import yaml
+            except ImportError:
+                return self._read_config_key_without_yaml(config_path, key)
             with open(config_path, "r") as f:
```

Ruff (0.15.20)

[warning] Unnecessary mode argument

Remove mode argument

(UP015)